### DISCUSSION TITLE
Marking rows as `is_deleted` when records no longer exist in upsert mode?

### DISCUSSION THREAD

**Discussion by nicklausroach** [slingdata-io/sling-cli#445] [Q&A]
I'm sure this has been discussed, but is there a way currently to handle the situation where the source data has removed rows in a table that is being ingested via upsert? I suppose this starts to dip into CDC world, but being able to add an `is_deleted` flag to the data would be extremely valuable. I imagine the only way around this currently is snapshotting.

**Comment by nicklausroach** [comment]
@flarco bumping this in case you missed it. Would love your thoughts!

**Reply by flarco** [reply]
Hey @nicklausroach, check out my answer here: https://github.com/slingdata-io/sling-cli/issues/333
